Assignment/Assignment/SampleData.cs

```diff
@@ -1,30 +1,94 @@
     {
-        // 1.
-        public IEnumerable<string> CsvRows => throw new NotImplementedException();
+        private List<string> _CsvRows;
 
-        // 2.
-        public IEnumerable<string> GetUniqueSortedListOfStatesGivenCsvRows() 
-            => throw new NotImplementedException();
+        public SampleData(string filePath)
+        {
 
-        // 3.
-        public string GetAggregateSortedListOfStatesUsingCsvRows()
-            => throw new NotImplementedException();
+            if (!File.Exists(filePath))//case if file does not exist
+            {
+                throw new FileNotFoundException("Error, file does not exist");
+            }
+            if (new FileInfo(filePath).Length == 0)//case if file passed in is empty
+            {
+                throw new ArgumentNullException("Error, file is empty");
+            }
 
-        // 4.
-        public IEnumerable<IPerson> People => throw new NotImplementedException();
+            try
+            {
+                _CsvRows = File.ReadAllLines(filePath).Skip(1).ToList();
+            }catch (Exception)
+            {
+                throw new ArgumentNullException("could not read from file");
+            }
+            
+            if (_CsvRows is null)
+            {
+                throw new ArgumentNullException("Error, reading lines from file went bad"); ;
+            }
+        }
 
-        // 5.
-        public IEnumerable<(string FirstName, string LastName)> FilterByEmailAddress(
-            Predicate<string> filter) => throw new NotImplementedException();
+        // 1.->Finished
+        public IEnumerable<string> CsvRows { get { return _CsvRows; } }
 
-        // 6.
+        // 2.->Finished
+        public IEnumerable<string> GetUniqueSortedListOfStatesGivenCsvRows()
+        {
+            List<string> states = new List<string>();
+            
+            states = CsvRows.Select(item => item.Split(",")).OrderBy(item => item[6]).Select(item => item[6]).Distinct().ToList();
+            states.Zip(states, (current, next) => string.Compare(current, next) <= 0).All(x => x);
+            return states;
+        }
+
+        // 3.->Finished
+        public string GetAggregateSortedListOfStatesUsingCsvRows() 
+        {
+            string statesString = "";
+            IEnumerable<string> statesList = GetUniqueSortedListOfStatesGivenCsvRows();
+            statesString = statesList.Aggregate((state1, state2) => state1 + ", " + state2);
+            return statesString;
+        }
+
+        // 4.->Finished
+        public IEnumerable<IPerson> People { get {
+                IEnumerable<IPerson> people =
+                    CsvRows.Select(line => line.Split(",")).OrderBy(line => line[5]).ThenBy(line => line[6]).ThenBy(line => line[7])
+                    .Select(item => new Person(item[1], item[2], new Address(item[4], item[5], item[6], item[7]), item[3])).ToList();
+                return people;
+            }

```

Comment:
```suggestion
        public IEnumerable<IPerson> People { get => CsvRows
                .Select(line => line.Split(","))
                .OrderBy(line => line[5])
                .ThenBy(line => line[6])
                .ThenBy(line => line[7])
                .Select(item => new Person(
                    item[1],
                    item[2],
                    new Address(
                        item[4],
                        item[5],
                        item[6],
                        item[7]),
                    item[3]));
```
You should consider writing long LINQ chains on multiple lines for better readability.

Additionally, you are sorting by city then state then zip. This should sort by State then city then zip.